### PR TITLE
Fixed bounded sequences JSON ParseTest

### PR DESCRIPTION
**Pull request by jmmorato**
Changing the `VreadVwriteTest.idl` MySeq definition to use a bounded sequence (`typedef sequence<short, 7> MySeq;`) makes the test failing due to an unhandled exception.

### CODE REVIEW

**Comment by jrw972** [comment]
@jmmorato Do you think the bug is in `grow`?

**Comment by mitza-oci** [comment]
`grow` shouldn't be called on bounded sequences, or should be extended to allow this

**Comment by jmmorato** [comment]
```cpp
if (sequence->unbounded()) {
  be_global->impl_ << indent << " OpenDDS::DCPS::grow(" << expression << ");\n";
} else {
  be_global->impl_ << indent << "  const ACE_CDR::ULong len = " << expression << ".length();\n";
  be_global->impl_ << indent << "  " << expression << ".length(len + 1);\n";
}
```

better like this?

File: dds/idl/value_reader_generator.cpp [hunk 1]
```diff
@@ -116,7 +116,12 @@ namespace {
     if (use_cxx11) {
       be_global->impl_ << indent << "  " << expression << ".resize(" << expression << ".size() + 1);\n";
     } else {
-      be_global->impl_ << indent << "  OpenDDS::DCPS::grow(" << expression << ");\n";
+      if (sequence->unbounded()) {
+        be_global->impl_ << indent << " OpenDDS::DCPS::grow(" << expression << ");\n";
+      } else {
+        be_global->impl_ << indent << "  const ACE_CDR::ULong len = " << expression << ".length();\n";
+        be_global->impl_ << indent << "  " << expression << ".length(len + 1);\n";
```
[the remark this answers is not in the document]
Comment (jrw972) [reply dds/idl/value_reader_generator.cpp:123]: It either needs to 1) return `false` or 2) skip the values until `value_reader.elements_remaining()` returns `false`.  I suggest just returning `false`.  Someday, we will rewrite/replace `vread/vwrite` with things from XTypes and DynamicData at which point we can do something smarter like try construct.

[the remark this answers is not in the document]
Comment (jmmorato) [reply dds/idl/value_reader_generator.cpp:124]: Implemented option 1. Check my last push.

Code generated example:
```cpp
if (i >= value.seq1.maximum()){
  return false;
}
const ACE_CDR::ULong len = value.seq1.length();
value.seq1.length(len + 1);
if (!value_reader.begin_element()) return false;
if (!value_reader.read_int16(value.seq1[i])) return false;
if (!value_reader.end_element()) return false;
```

**Review by jrw972 — APPROVED** [review]

**Comment by jrw972** [comment]
@jmmorato One more request:  Can you create a news fragment in `docs/news.d` so your contribution appears in the news for the next release?

**Review by jrw972 — APPROVED** [review]

**Pull request merged**